src/Microsoft.Identity.Web.TokenCache/IMsalTokenCacheProvider.cs

```diff
@@ -26,6 +26,13 @@ public interface IMsalTokenCacheProvider
         /// <param name="tokenCache">Token cache for which to initialize the serialization.</param>
         void Initialize(ITokenCache tokenCache);
 
+        /// <summary>
+        /// Safely initializes a token cache (which can be a user token cache or an app token cache).
+        /// </summary>
+        /// <param name="tokenCache">Token cache for which to initialize the serialization.</param>
+        /// <remarks>The safe terminology denotes to not propagating exceptions from caching layer to service, instead fallbacking to eSTS call.</remarks>
+        void SafeInitialize(ITokenCache tokenCache);
```

Comment:
I had to read the implementation to understand what this does. I would name it more along the lines of "no exceptions"